drivers/counter/Kconfig.mspm0gxxxx

```
@@ -0,0 +1,5 @@
+config COUNTER_MSPM0GXXXX
```

Should depend on the driver being enabled in the device tree as well
```suggestion
config COUNTER_MSPM0GXXXX
	depends on DT_HAS_TI_COUNTER_MSPM0_ENABLED
```